### PR TITLE
Avatar and Motto in profile page

### PR DESCRIPTION
**Pull request by MartinaVilimova**
- refactor avatar component
- add an avatar to the profile page
- create FormTextArea component
- create Motto component

### CODE REVIEW

File: src/pages/Profile.tsx
Comment (agnes97) [inline src/pages/Profile.tsx:28]: This is a huge problem. :sweat_smile: After Registration refactoring/errors were merged (now on master), userData file was DELETED. Profile nickname is also set the minute you register and e-mail is also taken from our own db so there was no reason to use them anymore.

We don't want to fetch data from random API when we already have our own database, **I understood this issue as giving users avatar field in Firebase firestore database and fetching it from our own data**.

Either way, your avatar branch can't be approved and merged when it uses data we no longer have. I'd recommend rebasing your branch on master and going from there.

File: src/components/Avatar/styled.ts
Comment (agnes97) [inline src/components/Avatar/styled.ts:4]: This looks good but sometimes I get black "border" between image and it's white border. I don't know why, maybe it's because of messages we're fetching from foreign API.
avatar

File: src/components/Form/styled.ts
Comment (agnes97) [inline src/components/Form/styled.ts:119]: Is text area always 12em in height? :thinking: If we only have around 3 rows of text allowed (maybe limit it with number of letters?), then it's confusing. User thinks he can add this big 12em paragraph but instead he's allowed only few sentences. 

In the end with this, avatar is too big (get blurry, recommended avatar width/height is at max. 200px, this is almost 250px) and after submitting motto, we get huge "gaps" between nickname and motto when it's short.

> Default velikosti souboru je něco kolem 200kb nebo velikostně něco jako 120 - 200px.

avatar2

File: src/components/Profile/styled.ts
Comment (agnes97) [inline src/components/Profile/styled.ts:95]: Combination of nickname and motto is weirdly centered.

center

File: src/components/Profile/Motto/index.tsx
Comment (agnes97) [inline src/components/Profile/Motto/index.tsx:32]: I can't see this placeholder unless I delete "suggested" motto "Člověk nežije proto, aby pracoval, ale pracuje, aby žil."

File: src/components/Profile/Motto/index.tsx
Comment (agnes97) [inline src/components/Profile/Motto/index.tsx:42]: In case of no motto, I'd add something like * * * (centered). Any text looks like we're forcing them to put some motto in while stars of some kind of "divider" works even without any text present.

File: src/components/Profile/styled.ts
Comment (agnes97) [inline src/components/Profile/styled.ts:101]: Cool styling! <3

File: src/pages/Profile.tsx
Comment (agnes97) [inline src/pages/Profile.tsx:25]: Part of this issue is also adding "default" avatar in case of no avatar. I haven't see that?

File: src/pages/Profile.tsx
Comment (MartinaVilimova) [reply src/pages/Profile.tsx:28]: Ok, so i have to adjust it. I didn't realize the random user would be gone.

File: src/components/Avatar/styled.ts
Comment (MartinaVilimova) [reply src/components/Avatar/styled.ts:4]: I know what you mean and I have no idea how to get rid of it. I tried different ways and none of them solved this problem for me. I'll try to think again. Another option to remove the border 😅

Comment (agnes97) [reply src/components/Avatar/styled.ts:4]: Maybe uploading our own avatars will solve the issue. (They might be uploading some images as a circle with border? While we are uploading rectangular images and setting them as cover.)

File: src/components/Form/styled.ts
Comment (MartinaVilimova) [reply src/components/Form/styled.ts:119]: Textarea has no restrictions yet and 12em is high due to the "Something about me" section. I can set other styles for the motto and other styles elsewhere if we really want to.

I don't understand the second paragraph very well 🤔 I don't understand the second paragraph very well. I reduce the image to 200px and I don't understand the gap.

File: src/components/Profile/styled.ts
Comment (MartinaVilimova) [reply src/components/Profile/styled.ts:95]: I don't see a problem🤔 the lines take up as much space as needed and we don't want the name at the very top.

File: src/components/Form/styled.ts
Comment (agnes97) [reply src/components/Form/styled.ts:119]: This is what I meant. In current case, each gap (margin or padding) is different. size. 

Name is pinned down:
current one

While motto is centered:
centered

Resulting in three different gaps:
každájiná

While this is what happens if you don't vertically center motto:
somethinglikethis

File: src/components/Profile/styled.ts
Comment (agnes97) [reply src/components/Profile/styled.ts:95]: This is what I meant. In current case, each gap (margin or padding) is different. size. 

Name is pinned down:
current one

While motto is centered:
centered

Resulting in three different gaps:
každájiná

While this is what happens if you don't vertically center motto:
somethinglikethis

Motto should be three rows of text max (something like that in number of letters), your first example should be forbidden.

File: src/components/Form/styled.ts
Comment (agnes97) [reply src/components/Form/styled.ts:119]: Well, we have to restrict this (Props?) or it'll overflow. Allowing users to put there any ammount of text completely destroys the idea of "**short first impression** motto". Making it short is also necessary for it to fit next to the avatar without breaking the design at some point. 

Gap explained in comment about centering. It should be horizontally centered (left and right), I didn't like how middle vertical gap centeres it (also in another comment with screenshots).

File: src/components/Profile/Motto/index.tsx
Comment (MartinaVilimova) [reply src/components/Profile/Motto/index.tsx:32]: Hm, but at first the user sees only the placeholder and then sees only his motto. we no longer need to see the placeholder. The proposed text is just an example of a motto.

But thanks to your comment, I realized that if the user entered some text (heslo, něco o mně atd.), it would be better to show the text than the textarea! The text area should only appear for the first time if nothing is filled. 🤔

I'll definitely take a look at that, good point 👍

File: src/pages/Profile.tsx
Comment (MartinaVilimova) [reply src/pages/Profile.tsx:25]: Yes. I have an icon when the picture doesn't appear.

Comment (MartinaVilimova) [reply src/pages/Profile.tsx:25]: .. just delete the access to the random user.
image

File: src/components/Profile/Motto/index.tsx
Comment (MartinaVilimova) [reply src/components/Profile/Motto/index.tsx:42]: ok 👍

File: src/pages/Profile.tsx
Comment (agnes97) [reply src/pages/Profile.tsx:25]: Awesome! :3 Sorry, I somehow overlooked it.

**Review by agnes97 — APPROVED** [review]

**Pull request merged**
